### PR TITLE
Use dense formatting for requirement version specifiers in diagnostics

### PR DESCRIPTION
**Pull request by zanieb**
For consistency with the output in "no solution" errors

### CODE REVIEW

File: crates/uv/tests/pip_check.rs [hunk 1]
```diff
@@ -206,8 +206,8 @@ fn check_multiple_incompatible_packages() -> Result<()> {
      + idna==2.4
      - urllib3==2.1.0
      + urllib3==1.20
-    warning: The package `requests` requires `idna <4, >=2.5`, but `2.4` is installed.
-    warning: The package `requests` requires `urllib3 <3, >=1.21.1`, but `1.20` is installed.
+    warning: The package `requests` requires `idna<4,>=2.5`, but `2.4` is installed.
+    warning: The package `requests` requires `urllib3<3,>=1.21.1`, but `1.20` is installed.
```
Comment (konstin) [inline crates/uv/tests/pip_check.rs:210]: I find this one hard to read, should split name and specifier?

**Review by konstin — APPROVED** [review]

File: crates/uv/tests/pip_check.rs (hunk 1 above)
Comment (zanieb) [reply crates/uv/tests/pip_check.rs:210]: The reason we don't tend to split the name and specifier is that we don't use "`" so it's less ambiguous as a single statement. I'm thinking we should probably remove the "`" here for consistency with our other messaging, but it's not really a priority. I just want us to have a consistent presentation.

Another option is we use colors to separate the specifier from the name i.e. bold names.

Comment (konstin) [reply crates/uv/tests/pip_check.rs:210]: i'd love using color for this, though still a space after the name for CI reports probably

Comment (zanieb) [reply crates/uv/tests/pip_check.rs:210]: I'll look into that

**Pull request merged**